Code/Framework/AzFramework/Tests/Spawnable/SpawnableTests.cpp

```diff
@@ -0,0 +1,513 @@
+              Spawnable::EntityAliasType::Original, Spawnable::EntityAliasType::Original, Spawnable::EntityAliasType::Original,
+              Spawnable::EntityAliasType::Original, Spawnable::EntityAliasType::Original });
+
+        AzFramework::Spawnable::EntityAliasVisitor visitor = m_spawnable->TryGetAliases();
+        ASSERT_TRUE(visitor.IsValid());
+
+        EXPECT_EQ(0, AZStd::distance(visitor.begin(), visitor.end()));
+    }
+
+    TEST_F(SpawnableTest, EntityAliasVisitor_Optimize_MixedOriginals_AllOriginalsRemoved)
+    {
+        using namespace AzFramework;
+        InsertEntities(8);
+        InsertEntityAliases<8>(
+            { 0, 0, 0, 1, 1, 2, 2, 2 }, { 0, 1, 2, 3, 4, 5, 6, 7 },
+            { Spawnable::EntityAliasType::Original, Spawnable::EntityAliasType::Original, Spawnable::EntityAliasType::Original,
+              Spawnable::EntityAliasType::Original, Spawnable::EntityAliasType::Disable, Spawnable::EntityAliasType::Original,
+              Spawnable::EntityAliasType::Replace, Spawnable::EntityAliasType::Original });
+
+        AzFramework::Spawnable::EntityAliasVisitor visitor = m_spawnable->TryGetAliases();
+        ASSERT_TRUE(visitor.IsValid());
+
+        EXPECT_EQ(2, AZStd::distance(visitor.begin(), visitor.end()));
+        EXPECT_EQ(Spawnable::EntityAliasType::Disable, visitor.begin()->m_aliasType);
+        EXPECT_EQ(Spawnable::EntityAliasType::Replace, visitor.begin()[1].m_aliasType);
+    }
+
+    TEST_F(SpawnableTest, EntityAliasVisitor_Optimize_MergeAfterOriginal_NoAdditionalOriginalIsInserted)
+    {
+        using namespace AzFramework;
+        InsertEntities(8);
+        InsertEntityAliases<8>(
+            { 0, 0, 1, 1, 2, 2, 2, 2 }, { 0, 1, 2, 3, 4, 5, 6, 7 },
+            { Spawnable::EntityAliasType::Original, Spawnable::EntityAliasType::Merge, Spawnable::EntityAliasType::Replace,
+              Spawnable::EntityAliasType::Merge, Spawnable::EntityAliasType::Original, Spawnable::EntityAliasType::Original,
+              Spawnable::EntityAliasType::Original, Spawnable::EntityAliasType::Original });
+
+        AzFramework::Spawnable::EntityAliasVisitor visitor = m_spawnable->TryGetAliases();
+        ASSERT_TRUE(visitor.IsValid());
+
+        EXPECT_EQ(4, AZStd::distance(visitor.begin(), visitor.end()));
+        EXPECT_EQ(Spawnable::EntityAliasType::Original, visitor.begin()->m_aliasType);
+        EXPECT_EQ(Spawnable::EntityAliasType::Merge, visitor.begin()[1].m_aliasType);
+        EXPECT_EQ(Spawnable::EntityAliasType::Replace, visitor.begin()[2].m_aliasType);
+        EXPECT_EQ(Spawnable::EntityAliasType::Merge, visitor.begin()[3].m_aliasType);
+    }
+
+
+    //
+    // UpdateAliasType
+    //
+
+    TEST_F(SpawnableTest, EntityAliasVisitor_UpdateAliasType_AllToOriginal_NoAliasesAfterOptimization)
+    {
+        using namespace AzFramework;
+        InsertEntities(8);
+        InsertEntityAliases<8>(
```

Comment:
repeated nit: Can these be replaced with 'DefaultEntityAliasTestCount' as well?

Reply:
I'd rather not because those each need to provide arguments with exactly that number of entries. Having the exact number there makes it more readable imho than have `DefaultEntityAliasTestCount`.